routingpy/routers/opentripplanner_v2.py

```diff
@@ -165,21 +135,18 @@ def directions(
         response = self.client._request(
             "/otp/routers/default/index/graphql", post_params=params, dry_run=dry_run
         )
-        return self._parse_directions_response(response, num_itineraries)
-
-    def _timestamp_to_utc_datetime(self, timestamp):
-        dt = datetime.datetime.fromtimestamp(timestamp / 1000)
-        return dt.astimezone(datetime.timezone.utc)
+        return self.parse_directions_response(response, num_itineraries)
 
-    def _parse_directions_response(self, response, num_itineraries):
+    @staticmethod
+    def parse_directions_response(response, num_itineraries):
         if response is None:  # pragma: no cover
             return Directions() if num_itineraries > 1 else Direction()
 
         directions = []
         for itinerary in response["data"]["plan"]["itineraries"]:
-            distance, geometry = self._parse_legs(itinerary["legs"])
-            departure_datetime = self._timestamp_to_utc_datetime(itinerary["startTime"])
-            arrival_datetime = self._timestamp_to_utc_datetime(itinerary["endTime"])
+            distance, geometry = OpenTripPlannerV2._parse_legs(itinerary["legs"])
+            departure_datetime = convert.timestamp_to_tz_datetime(itinerary["startTime"], "UTC")
+            arrival_datetime = convert.timestamp_to_tz_datetime(itinerary["endTime"], "UTC")
```

Comment:
this I don't really understand @khamaileon : does that mean that in the OTP result there's no info which timezone the destination (or departure) is referring to? Is there no way to get the timezone info back from OTP? If not, I'd suggest to find the timezone similar to valhalla with `timezonefinder`, otherwise anyone using this has the burden to look up the timezone to get the local time.

Reply:
I think we can consider the input timezone as the output timezone. That's why I insisted on a datetime aware input.

Reply:
It expects a local date & time in the request. But we ask the user for an aware datetime so we can use the same timezone to transform the output. Maybe I'm wrong, but I find it hard to imagine a user requesting an itinerary on one timezone with a result on another timezone.

BTW, there is also a way to retrieve the timezone for transit lges via agency/timezone.

Reply:
> I find it hard to imagine a user requesting an itinerary on one timezone with a result on another timezone.

That's actually quite common for car or long-distance transit (e.g. Amtrak crossing the US).

> BTW, there is a way to retrieve the timezone for transit legs via agency/timezone.

I think that's not granular enough. We'd need the stops.txt/stop_timezone (if OTP exposes that) and I wouldn't be sure that's consistently handled by all operators who indeed have stops in different timezones. 

FWIW, I asked a OTP core dev offline, maybe he has some more insight/recommendations.

Reply:
Haha he sent me your matrix questions and his answers. So it's the same situation for OTP as we have in Valhalla. It's a rare situation that one operator has timezone-crossing stops, but still it's common to cross timezone in one itinerary. I'd use the same approach for now as in Valhalla and look it up dynamically by its location with `timezonefinder`. Apparently they want to change it in OTP similar to what I'd like in Valhalla, so this would hopefully only be a temp solution. Ok for you?

Reply:
> That's actually quite common for car or long-distance transit (e.g. Amtrak crossing the US).

Yeah, I see. I think I was too focused on public transport where you only have one timezone per network.

> I think that's not granular enough. We'd need the stops.txt/stop_timezone (if OTP exposes that) and I wouldn't be sure that's consistently handled by all operators who indeed have stops in different timezones.

OTP exposes timezone a the stop level, but maybe that's too much information to bring up.

> FWIW, I asked a OTP core dev offline (Cc @leonardehrenfried), maybe he has some more insight/recommendations.

Lol I had asked the question on the OTP chat and it was him who answered me!

Reply:
Yes I'm ok with it! Don't forget timezonefinder has a cost (near 1s / 10000 query)

Reply:
Come to think of it, now that we won't be using the input timezone for output, do we still need an input datetime aware? And what happens if we enter a departure timezone that isn't the one at the point of origin? Sorry, I can make the changes if we change.

Reply:
yeah right, it’s a bit of cost, but I guess it’s drowning well in the OTP/Valhalla response time:)

God I hate datetime stuff! It always the head quite a bit.. I’m not sure about OTP‘s isochrones, but I’m now fairly sure that expects a local time as well (timezone would be meaningless). If that’s the case, all routers only need the local time and we can ditch tz aware input? The output is probably the only thing that should have timezone awareness. Google does it right, OTP & Valhalla hopefully soon too.